# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineCodeGen.cpp
# Commit: d9c1e44eb0ad
# Full Hash: d9c1e44eb0addfc3e44ae299c86381128aa08e7f
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-15 21:49:48
# Regressor Bug: 1601599
# File Overlap Count: 1
# Description:
#   Bug 1601599 part 4 - Support Ion OSR at all loops. r=tcampbell
#   
#   Depends on D56700
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56701
# ==============================================================================

diff -r 93842c3971f8 -r d9c1e44eb0ad js/src/jit/BaselineCodeGen.cpp
--- a/js/src/jit/BaselineCodeGen.cpp	Sun Dec 15 11:33:36 2019 +0000
+++ b/js/src/jit/BaselineCodeGen.cpp	Sun Dec 15 11:34:34 2019 +0000
@@ -1315,11 +1315,6 @@
     if (handler.analysis().info(pc).loopHeadInCatchOrFinally) {
       return true;
     }
-
-    if (!LoopHeadCanIonOsr(pc)) {
-      // OSR into Ion not possible at this loop.
-      return true;
-    }
   }
 
   Label done;